src/command/manifest/manifest-command.ts

```diff
@@ -53,25 +87,91 @@ export class ManifestCommand extends RootCommand {
     return new TextEncoder().encode(path)
   }
 
-  protected async initializeNode(reference: string): Promise<MantarayNode> {
-    const manifest = await this.bee.downloadData(reference)
-    const node = new MantarayNode()
-    node.deserialize(manifest)
-    await loadAllNodes(this.load.bind(this), node)
+  protected decodeReference(reference: Reference | Uint8Array): string {
+    return Buffer.from(reference).toString('hex')
+  }
 
-    return node
+  protected async initializeNode(hash: string, path?: string | null): Promise<NodeSearchResult> {
+    try {
+      const searchResult = await this.getDeserializedNode(hash, path)
+
+      if (!searchResult) {
+        if (path) {
+          throw new Error('Could not deserialize or find Mantaray node for reference ' + hash + ' and path ' + path)
+        }
+        throw new Error('Could not deserialize or find Mantaray node for reference ' + hash)
+      }
+      await loadAllNodes(this.load.bind(this), searchResult.node)
+
+      return searchResult
+    } catch (error: unknown) {
+      // FIXME in mantaray-js
+      if (Reflect.get(error as Record<string, unknown>, 'message') === 'Wrong mantaray version') {
+        Printer.error('The reference provided is not a root manifest hash')
+        exit(1)
+      } else {
+        throw error
+      }
+    }
   }
 
   protected async saveAndPrintNode(node: MantarayNode, stamp: string): Promise<void> {
     const reference = await node.save(this.createSaver(stamp))
     this.resultHash = (reference as Buffer).toString('hex')
     this.console.log(this.resultHash)
   }
+
+  private async getNodeForReference(reference: string): Promise<MantarayNode> {
+    const data = await this.bee.downloadData(reference)
+    const node = new MantarayNode()
+    node.deserialize(data)
+
+    return node
+  }
+
+  private async getDeserializedNode(reference: string, prefix?: string | null): Promise<NodeSearchResult | null> {
+    if (prefix) {
+      return this.findNodeForPrefix(reference, prefix)
+    } else {
+      const manifest = await this.bee.downloadData(reference)
+      const node = new MantarayNode()
+      node.deserialize(manifest)
+
+      return { node, prefix: '' }
+    }
+  }
+
+  private async findNodeForPrefix(
+    reference: string,
+    prefix: string,
+    currentPath = '',
+  ): Promise<NodeSearchResult | null> {
+    const node = await this.getNodeForReference(reference)
+
+    if (!node.forks) {
+      return null
+    }
+
+    for (const fork of Object.values(node.forks)) {
+      const path = currentPath + Buffer.from(fork.prefix).toString('utf-8')
+
+      if (path.startsWith(prefix)) {
+        const reference = this.decodeReference(fork.node.getEntry as Reference)
+        const match = await this.getNodeForReference(reference)
+
+        return { node: match, prefix: path }
+      } else if (prefix.startsWith(path)) {
+        return this.findNodeForPrefix(this.decodeReference(fork.node.getEntry as Reference), prefix, path)
```

Comment:
the first parameter of `findNodeForPrefix` is already decoded in `reference` constant.